modules/passkey/test/SafeWebAuthnSignerFactory.spec.ts

```diff
@@ -0,0 +1,178 @@
+    })
+
+    it('Should return different signer for different inputs', async () => {
+      const { factory, mockVerifier } = await setupTests()
+
+      const x = ethers.id('publicKey.x')
+      const y = ethers.id('publicKey.y')
+
+      const signer = await factory.getSigner(x, y, mockVerifier)
+
+      for (const params of [
+        [ethers.id('publicKey.otherX'), y, mockVerifier],
+        [x, ethers.id('publicKey.otherY'), mockVerifier],
+        [x, y, '0x0000000000000000000000000000000000000100'],
+      ] as const) {
+        expect(await factory.getSigner(...params)).to.not.equal(signer)
+      }
+    })
+  })
+
+  describe('createSigner', function () {
+    it('Should create a signer and return its address', async () => {
+      const { factory, mockVerifier } = await setupTests()
+
+      const x = ethers.id('publicKey.x')
+      const y = ethers.id('publicKey.y')
+
+      const signer = await factory.createSigner.staticCall(x, y, mockVerifier)
+
+      expect(ethers.dataLength(await ethers.provider.getCode(signer))).to.equal(0)
+
+      await factory.createSigner(x, y, mockVerifier)
+
+      expect(ethers.dataLength(await ethers.provider.getCode(signer))).to.not.equal(0)
+    })
+
+    it('Should be idempotent', async () => {
+      const { factory, mockVerifier } = await setupTests()
+
+      const x = ethers.id('publicKey.x')
+      const y = ethers.id('publicKey.y')
+
+      const signer = await factory.createSigner.staticCall(x, y, mockVerifier)
+
+      await factory.createSigner(x, y, mockVerifier)
+
+      expect(await factory.createSigner.staticCall(x, y, mockVerifier)).to.eq(signer)
+      await expect(factory.createSigner(x, y, mockVerifier)).to.not.be.reverted
+    })
+  })
+
+  describe('isValidSignatureForSigner', function () {
+    it('Should return true when the verifier returns true', async () => {
+      const { factory, mockVerifier } = await setupTests()
+
+      const dataHash = ethers.id('some data to sign')
+      const clientData = {
+        type: 'webauthn.get' as const,
+        challenge: base64UrlEncode(dataHash),
+        origin: 'https://safe.global',
+      }
+
+      const r = ethers.id('signature.r')
+      const s = ethers.id('signature.s')
+      const x = ethers.id('publicKey.x')
+      const y = ethers.id('publicKey.y')
+
+      const signature = getSignatureBytes({
+        authenticatorData: DUMMY_AUTHENTICATOR_DATA,
+        clientDataFields: '"origin":"https://safe.global"',
+        r,
+        s,
+      })
+
+      await mockVerifier.givenCalldataReturnBool(
+        ethers.solidityPacked(
+          ['bytes32', 'uint256', 'uint256', 'uint256', 'uint256'],
+          [ethers.sha256(encodeWebAuthnSigningMessage(clientData, DUMMY_AUTHENTICATOR_DATA)), r, s, x, y],
+        ),
+        true,
+      )
+
+      expect(await factory.isValidSignatureForSigner(dataHash, signature, x, y, mockVerifier)).to.equal('0x1626ba7e')
```

Comment:
Can we use the magic value here as well?

Reply:
This gets addressed in #374 already